third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -0,0 +1,27 @@
+#include "TritonAMDGPUTransforms/Passes.h"
+#include "TritonAMDGPUTransforms/PointerCanonicalizer.h"
+#include "mlir/Pass/Pass.h"
+
+using namespace mlir;
+
+#define GEN_PASS_CLASSES
+#include "TritonAMDGPUTransforms/Passes.h.inc"
+
+// This pass is simply calling the fat pointer canonicalization utility
+// on the given MLIR module
+class TritonAMDGPUCanonicalizePointersPass
+    : public TritonAMDGPUCanonicalizePointersBase<
+          TritonAMDGPUCanonicalizePointersPass> {
+public:
+  TritonAMDGPUCanonicalizePointersPass() = default;
+
+  void runOnOperation() override {
+    ModuleOp m = getOperation();
```

Comment:
Theoretically, because the pointer updates could propagate across functions (although this is not implemented yet). I mean if I have something like:
```
%ptr0 = addptr %ptr, %offset
tt.call "foo"(%ptr0)
```
I can still transform this into:
```
tt.call "foo"(%ptr, %offset)
```
And then have the `addptr` closer to the load.